api/v1alpha1/retry_types.go

```diff
@@ -0,0 +1,111 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Retry defines the retry strategy to be applied.
+type Retry struct {
+	// NumRetries is the number of retries to be attempted. Defaults to 2.
```

Comment:
Is there a particular reason why the default retry is different from the envoy default(1)?

Reply:
We're using [istio defaults](https://github.com/istio/istio/blob/b652ff114d85c2ed15034e1c3c87447d96b9c769/pilot/pkg/networking/core/v1alpha3/route/retry/retry.go#L36) as a real-world example for reasonable defaults here.